### PR TITLE
🔐 Restrict OAuth Client Creation to Superusers with Feature Flag

### LINKED ISSUE #2104: feature flag OUT the oauth client creation. for now, only allow `superuser` to create clients.

### PR DESCRIPTION
**Pull request by elias-ba**
## Related issue

Re #2102 
Fixes #2104

## Review checklist

- [x] I have performed a **self-review** of my code
- [x] I have verified that all appropriate **authorization policies** (`:owner`, `:admin`, `:editor`, `:viewer`) have been implemented and tested
- [ ] If needed, I have updated the **changelog**
- [ ] Product has **QA'd** this feature

### CODE REVIEW

**Review by taylordowns2000 — CHANGES_REQUESTED** [review]
1. looks like oauth client creation is still possible from the project credentials interface, though the button has been altered (image 1)

image

File: lib/lightning_web/live/components/credentials.ex [hunk 1]
```diff
@@ -323,18 +324,7 @@ defmodule LightningWeb.Components.Credentials do
...
-          fill="currentColor"
-          aria-hidden="true"
-        >
-          <path
-            fill-rule="evenodd"
-            d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
-            clip-rule="evenodd"
-          />
-        </svg>
+        <Heroicons.chevron_down class="ml-1 h-5 w-5" />
```
Comment (stuartc) [inline lib/lightning_web/live/components/credentials.ex:327]: Nice, thank you for removing inline svgs 🙌

File: lib/lightning_web/live/components/credentials.ex [hunk 2]
```diff
@@ -357,7 +347,10 @@ defmodule LightningWeb.Components.Credentials do
               phx-click={show_modal(target)}
               disabled={@disabled}
             >
-              <%= name %>
+              <%= name %><span
+                :if={Map.get(option, :badge)}
+                class="ml-2 inline-flex items-center rounded-md bg-gray-50 px-1.5 py-0.5 text-xs font-medium text-gray-600 ring-1 ring-inset ring-gray-500/10"
+              ><%= Map.get(option, :badge) %></span>
             </a>
```
Comment (stuartc) [inline lib/lightning_web/live/components/credentials.ex:354]: I think this interface is kinda confused, this is best encapsulated using slots.

The `options` assign for creating n number of a tags is exactly where slots are best. 

Also I see the button is only rendered in one place, with one set of options. I don't see the need for it to be so generic vs a composed function component that reuses other function components (if any).

**Review by stuartc — CHANGES_REQUESTED** [review]
I see there are bunch of tests failing, but also the `options_menu_button` component is too complex given its use case here. I think make it a single component that takes no arguments or if there is some kind of other use case I'm not aware of, let me know and we can chat about how to leverage slots etc.

**Pull request merged**